hnn_core/dipole.py

```diff
@@ -198,15 +249,67 @@ def convert_fAm_to_nAm(self):
     def scale(self, fctr):
         for key in self.data.keys():
             self.data[key] *= fctr
-        return fctr
+        return self
 
-    def smooth(self, winsz):
-        # XXX: add check to make sure self.times is
-        # not smaller than winsz
-        if winsz <= 1:
-            return
-        for key in self.data.keys():
-            self.data[key] = _hammfilt(self.data[key], winsz)
+    def smooth(self, *, window_len=None, h_freq=None):
```

Comment:
What do you think about an `inplace=False` argument to enable chaining?

Reply:
I think I'd prefer `copy=True` as an option (`False` by default), if anything. Tomatos, tomatoes...